yang/vyatta-protocols-frr-isis-v1.yang

```diff
@@ -0,0 +1,822 @@
+        leaf metric-id {
+          type uint32 {
+            range "0..16777215";
+          }
+          default "0";
+          configd:help "Metric used for the redistributed route. If 0,
+          the default-metric attribute is used instead.";
+          description "Metric used for the redistributed route. If 0,
+          the default-metric attribute is used instead.";
+        }
+        leaf route-map {
+          type string {
+            length "1..max";
+          }
+          configd:help "Route map reference.";
+          description "Route map reference";
+        }
+      }			
+    }
+  }
+
+  grouping redistribute-default {
+    container always {
+      configd:help "Always advertise default route.";
+      description "Always advertise default route.";
+      uses redistribute-attributes;
+    }
+    uses redistribute-attributes;
+  }
+
+
+  grouping isis-net {
+    container default-information {
+      configd:help "Distribution of default information.";
+      description "Distribution of default information.";
+      container originate {
+        configd:help "Distribute a default route.";
+        description "Distribute a default route.";
+        list ipv4 {
+          key "level";
+          configd:help "Distribute default route for IPv4.";
+          description "Distribute default route for IPv4.";
+          leaf level {
+            type level;
+          }
+        }
+
+        list ipv6 {
+          key "level";
+          configd:help "Distribute default route for IPv6.";
+          description "Distribute default route for IPv6.";
+          leaf level {
+            type level;
+          }
+        }
+      }
+    }
+    leaf net {
+      type net-address;
+      configd:help "XX.XXXX. ... .XXX.XX  Network entity title (NET)";
+      description "List of OSI NET addresses for this protocol instance.";
+    }
+    leaf log-adjacency-changes {
+      type empty;
+      configd:help "Log changes to the IS-IS adjacencies in this area.";
+      description "Log changes to the IS-IS adjacencies in this area.";
+    }
+    container lsp-gen-interval {
+      configd:help "Minimum LSP regeneration interval.";
+      description "Minimum LSP regeneration interval.";
+      leaf level-1 {
+        type uint16 {
+          range "1..120";
+        }
+        units "seconds";
+        configd:help "Minimum time allowed before level-1 LSP retransmissions.";
+        description "Minimum time allowed before level-1 LSP retransmissions.";
+      }
+      leaf level-2 {
+        type uint16 {
+          range "1..120";
+        }
+        units "seconds";
+        configd:help "Minimum time allowed before level-2 LSP retransmissions.";
+        description "Minimum time allowed before level-2 LSP retransmissions.";
+      }
+    }
+    container lsp-refresh-interval {
+      description "LSP refresh interval";
+      configd:help "LSP refresh interval";
+      leaf level-1 {
+        type uint16;
```

Comment:
The isisd commands restrict these values to the range 1-65235. I'd recommend applying this range to the `level-1` and `level-2` leaves here; this will prevent a user from configuring a value unsupported by isisd.

Reply:
This is Resolved.